test/extensions/filters/http/cache/cache_filter_integration_test.cc

```diff
@@ -297,6 +297,348 @@ TEST_P(CacheIntegrationTest, GetRequestWithBodyAndTrailers) {
   }
 }
 
+TEST_P(CacheIntegrationTest, ServeHeadRequest) {
+  useAccessLog("%RESPONSE_FLAGS% %RESPONSE_CODE_DETAILS%");
+  // Set system time to cause Envoy's cached formatted time to match time on this thread.
+  simTime().setSystemTime(std::chrono::hours(1));
+  initializeFilter(default_config);
+
+  const Http::TestRequestHeaderMapImpl request_headers = {
```

Comment:
high level question on the test structure. There are a lot of common (or maybe mostly common) stanzas of code in all your new tests. Would it be worth factoring these out into common helper methods in the test-class? The benefit is that by factoring out the common boilerplate, it might be easier to scan over the tests and see what the differences are.

Reply:
I have refactored the tests slightly, nevertheless ensuring that I always EXPECT in the tests rather than in the helper methods to make it really clear what is being tested. Let me know what you think!

Reply:
This looks better, and I agree. Thank you!